File: Backend/API/src/main/java/com/example/localhistory/quiz/dto/response/PublicQuizDTO.java
package com.example.localhistory.quiz.dto.response;

import lombok.Data;

import java.util.List;

@Data
public class PublicQuizDTO {
    private Long id;
    private String title;
    private Long landmarkId;
    private boolean alreadyCompleted;
    private List<PublicQuizQuestionDTO> questions;
}

File: Frontend/app/src/main/java/com/example/localhistory/model/response/PublicQuizDTO.kt
package com.example.localhistory.model.response

data class PublicQuizDTO(
    val id: Long,
    val title: String,
    val landmarkId: Long,
    val alreadyCompleted: Boolean = false, // tells if the authenticated user has completed the quiz
    val questions: List<PublicQuizQuestionDTO>
)
